types/jest/index.d.ts

```diff
@@ -328,7 +328,7 @@ declare namespace jest {
      * @remarks
      * For mocking functions, and 'get' or 'set' accessors, use `jest.spyOn()` instead.
      */
-    function replaceProperty<T, K extends keyof T>(obj: T, key: K, value: T[K]): ReplaceProperty<T[K]>;
+    function replaceProperty<T extends {}, K extends keyof T>(obj: T, key: K, value: T[K]): ReplaceProperty<T[K]>;
```

Comment:
I'm not sure we want `{}` here: this means any non-nullish type.
```suggestion
    function replaceProperty<T extends object, K extends keyof T>(obj: T, key: K, value: T[K]): ReplaceProperty<T[K]>;
```

Reply:
I might be wrong on whether jest can override properties on primitives though. If it can then maybe we don't need this constraint.